ambry-api/src/main/java/com/github/ambry/config/ReplicationConfig.java

```diff
@@ -181,6 +181,17 @@
   @Default("")
   public final Set<String> replicationVcrRecoveryPartitions;
 
+  /**
+   * To specify the type of replication to be used for inter colo replication.
+   * It has two values - "all-to-all" or "leader-based". Default value is "all-to-all".
+   * If set to "leader-based", inter colo replication between will be limited to the leaders(as elected by Helix) of the partition of each datacenter.
+   * If set to "all-to-all", inter colo replication will be in an all-to-all fashion, i.e. each replica talks to all other replicas irrespective of their state.
+   * Intra colo replication will continue as all-to-all fashion in both the models.
+   */
+  @Config("replication.model.across.datacenters")
+  @Default("all-to-all")
```

Comment:
nit: I suggest create a static string variable for this one
```
public static final String DEFAULT_REPLICATION_MODEL_ACROSS_DATACENTRS = ReplicationModelType.ALL_TO_ALL.toString().toLowerCase();

@Config("replication.model.across.datacenters")
@Default(DEFAULT_REPLICATION_MODEL_ACROSS_DATACENTERS)
public final String replicationModelAcrossDatacenters;
```

Reply:
Sure, added a static string constant variable. I directly defined the variable name as - public static final String DEFAULT_REPLICATION_MODEL_ACROSS_DATACENTRS = "all-to-all". It wasn't accepting enum name as a constant variable in the @default annotation.

Reply:
You don't have to pass enum name to the @Default annotation. My idea is to make sure we have consistent string representation here. If we assign DEFAULT_REPLICATION_MODEL_ACROSS_DATACENTERS with the enum's lowercase string, then in ReplicationEngine you can just do
```
replicationModelType = ReplicationModelType.valueof(replicationConfig.replicationModelAcrossDatacenters.toUpperCase());
```